### ISSUE TITLE
Fork

### ISSUE THREAD

**Issue by MrNavaStar** [michiruf/MCInvSync#1]
Hey! Im glad your working to make improvements on this mod! I do have a few questions however:

Is there a reason you posted your fork on modrinth rather than making a pr to the offical repo? I am happy to accept feedback and make necessary changes. 

I don't totally understand your first point for the fork, I never got around to actually doing anything with that issue, hence why is was still open. I always push my changes to github before a release is made.

I am totally okay with you forking the project and making your own changes, however if your going to post the mod I would ask that you change the mod name and icon.

Thanks!

**Comment by michiruf** [comment]
Hi there, thanks for your request.

> Is there a reason you posted your fork on modrinth rather than making a pr to the offical repo? I am happy to accept feedback and make necessary changes.

Yeah there is, I definetly test all my stuff on my servers that I have with a bunch of friends to ensure maximum stability, so I need an upstream as fast as possible to do so. I can totally make you a PR when I can ensure stability when I've tested this with my friends on my own server, but you gotta be aware that every part of the code has changed.
If I've changed in a feature based approach, it would have been pretty clear to me to make a PR way earlier, but this is not just a feature improvement rather than a completely refactoring of everything.

---

> I don't totally understand your first point for the fork, I never got around to actually doing anything with that issue, hence why is was still open. I always push my changes to github before a release is made.

Yeah to be fair, I guess I messed up something there. The reason to believe that, was cause you weren't responding to a message in the given issue posted in my reasons for the fork as well as the fact, that another issue got resolved ([here](https://github.com/MrNavaStar/InvSync/issues/25)) not having changes in the repo, where in my understanding it should have been necessary. That got my assumption that you might have another upstream and forgot to push to github aswell, but assuming these things definetly was wrong and I apologizefor that! I will of cause immediately remove those reasons.

Additionally, I wanted to point out that I tried to release my stuff on modrinth without fork reasons at all and it got rejected like you could read in your Discord support thread, so pointing these things out wasn't my plan at all.

---

> I am totally okay with you forking the project and making your own changes, however if your going to post the mod I would ask that you change the mod name and icon.

The reason I used your icon was ment to go in the other direction: I especially wanted to credit you for your work and someone that might come across could see, that there is also your original mod, so the person could consider using your mod instead of my separate release.
Also to support your work I kept this [You can support MrNavaStar by buying him a coffee!](https://github.com/michiruf/MCInvSync/blob/44ec3d485d4375c6cd507c351099df03d5d535c8/README.md#why-this-fork).
What I should do with the icon is totall up to you, so if you stil wish that I shall remove that, I will do as soon as you say, like I said for me that just was to credit you even further and tell people to check out your stuff.

A side note from me is to not publish your icon with an MIT license, if you dont want other people to use it. I definetly checked your license even before forking and so thought everything should be fine.

When it comes to the name of the mod, I cannot agree to your request or the modrinth teams decision to force me to change that. As also stated in your modrinth support discussion, there has already been mods that have that name before you created yours. A few of them are:
* [Inventory Sync 2012](https://bukkit.org/threads/inventory-sync.113495/) uses the command `/invsync ...`
* [InvSync by Cancelcloud](https://www.spigotmc.org/resources/%C2%BB-invsync-%E2%9C%85-1-18-1-18-1.99780/)
* [InvSync by Compaszer](https://www.curseforge.com/minecraft/mc-mods/invsync)
* [Inventory Sync by BrainStone](https://ore.spongepowered.org/BrainStone/Inventory-Sync) uses rights starting with `invsync. ...`

The reason why there are plenty of mods called InvSync is, that this is rather a functionality description than a own name for a mod. If it was something like FeedTheBeast, I would happily accept to change the name, but when the name is deduced by its functionality (which I also did for all of my mods so far, because its definetly the easiest to find and easiest to understand name for a mod), there is no point in claiming this specific name. If anyone is about to publish a mod that makes Allays follow the player always, I will happily accept the fact, that this mod will have the same name as my AllayFollowAlways mod. Furthermore, opposed to the opinion of the modrinth team, I definetly think, that being in active development state does not change anything for having the right on a name or not, so either we both need to change the name, or none of us. 

Please let me know what you think about this.

---

To summarize things up:
* Shall I still remove the logo?
* Do you want a PR that changes everything?
* What do you think about the name as stated above?
* I apologize again for the misassumptions

Thanks for your time and work!

**Comment by MrNavaStar** [comment]
Thanks for getting back to me! I appreciate your response a lot. I think you make some good points here. 

First thing, based on what your saying I can see that a pr doesn't make a ton of sense. Normally I would probably accept it anyway, but I have a project in the works that will probably make InvSync Obsolete Anyway. 

I see your points about the name, and I would still prefer you use a different one to clear confusion on modrinth, but there is nothing I can actually do to prevent you from using it as it was never my name to begin with. 

In terms of the icon, I would appreciate if you can make your own. I probably should have picked a better license, but the icon is really the only thing that I would want a different license for. I think it helps to make it clear that the project is a fork and any improvements/issues are not associated with the original project. 

Lastly thank you for giving me credit on both GitHub and Modrinth, I appreciate that a lot. 

I really appreciate how open you have been and look forward to seeing what improvements your able to make to the project!

**Comment by MrNavaStar** [comment]
Oh and I just wanted to mention that I have since fixed the first two reasons you mentioned for forking. I must have had an issue on a push when I fixed the time.sleep thing, and you were right the change was never on github.  I have now fixed that and I have also fixed the database connection closed issue, it was really quite silly.

Anyway I will continue to work on fixing the issues you mentioned as your reason for forking, as I do see them as valid reasons. I originally wrote this mod 2+ years ago as a way to learn about sql, and I think its pretty obvious I had no clue what I was doing. The big one was the connections that get created and closed on every event, that certainly needs to be fixed.

If your willing, I would be very interested in a pr to sqlib to fix many of those issues, I image you could port a lot of your work over to it, but no pressure if thats outside of the scope of what your willing to work on. I use sqlib for a few other projects, hence why it is its own library,

**Comment by michiruf** [comment]
> First thing, based on what your saying I can see that a pr doesn't make a ton of sense. Normally I would probably accept it anyway, but I have a project in the works that will probably make InvSync Obsolete Anyway.

Sounds good :)

> I see your points about the name, and I would still prefer you use a different one to clear confusion on modrinth, but there is nothing I can actually do to prevent you from using it as it was never my name to begin with.

Ill change the title of the readme to MR InvSync aswell, so it should be pretty clear where the user is at.

> In terms of the icon, I would appreciate if you can make your own. I probably should have picked a better license, but the icon is really the only thing that I would want a different license for. I think it helps to make it clear that the project is a fork and any improvements/issues are not associated with the original project.

Okay sounds reasonable, Ill remove the icon then since I dont care about a icon.

> Lastly thank you for giving me credit on both GitHub and Modrinth, I appreciate that a lot.

Of course!

> I really appreciate how open you have been and look forward to seeing what improvements your able to make to the project!

When Ill figure out, how I can load a player from the files, I will try to make a command that will synchronize all player data from the server immediately.

> If your willing, I would be very interested in a pr to sqlib to fix many of those issues, I image you could port a lot of your work over to it, but no pressure if thats outside of the scope of what your willing to work on. I use sqlib for a few other projects, hence why it is its own library,

To be honest thats a little bit out of scope for me. Since I got the database handling with ORMLite working, you can of have a look at the code [here](https://github.com/michiruf/MCInvSync/tree/master/src/main/java/de/michiruf/invsync/data)
